app/uk/gov/hmrc/eventhub/service/PublisherService.scala

```diff
@@ -52,6 +56,15 @@ class PublisherService @Inject() (
       throw new Exception(s"Unknown Transaction error $e")
   }
 
+  private[service] def matchingPath(event: Event, subscribers: List[Subscriber]): Boolean = {
+    val subscriberWithPath = subscribers.find(_.pathFilter.nonEmpty).flatMap(_.pathFilter)
+    val noMatchingPath = subscriberWithPath match {
+      case Some(path) => path.read[net.minidev.json.JSONArray](Json.toJson(event).toString()).isEmpty
+      case _          => false
+    }
+    !noMatchingPath
+  }
+
```

Comment:
This function finds the first subscriber with a nonempty path and checks, all subscribers should be checked and only those that match should have the event published, the logic here looks wrong to me.